### PR TITLE
ADF-780 e2e items import export

### PR DESCRIPTION
**Pull request by meefox**
Related to https://oat-sa.atlassian.net/browse/ADF-780

1) Checkout tao-core https://github.com/oat-sa/tao-core/pull/3430
2) Checkout taoItems https://github.com/oat-sa/extension-tao-item/pull/572
3) Please run test against playground ENV
4) Run `npm run cy:open` from tao/view/builds
5) Execute spec taoItems/views/cypress/tests/properties/simple.spec.js
6) Observe 9 green cases
Screenshot 2022-02-25 at 14 58 44

### CODE REVIEW

File: views/cypress/tests/import-export.spec.js [hunk 1]
```diff
@@ -84,13 +98,65 @@ describe('Import/export items', () => {
...
+            format: 'APIP Content Package'
+        },{
+            format: 'QTI Metadata',
+        },{
+            format: 'RDF'
+        }].forEach((testcase, index) => {
+            it(`${index}: "Export - ${testcase.format || 'QTI Package 2.2'}"`, function () {
+                cy.clearDownloads();
+                cy.exportFromSelectedClass(selectors.exportItem, selectors.exportItemUrl, className, testcase.format);
+                cy.wait(50); // Safe delay for working with FS in Cypress
```
Comment (gabrielfs7) [inline views/cypress/tests/import-export.spec.js:135]: Isn't 50s too much?

File: views/cypress/tests/import-export.spec.js [hunk 2]
```diff
@@ -63,19 +65,31 @@ describe('Import/export items', () => {
...
+            format: 'QTI/APIP XML Item Document',
+            filename: 'e2e_test_item.xml'
+        },{
+            format: 'RDF',
+            filename: 'e2e_test_item.rdf'
+        }].forEach((testcase, index) => {
+            it(`${index}: "Import - ${testcase.format || 'QTI/APIP Content Package'}"`, function () {
+                cy.selectNode(selectors.root, selectors.itemClassForm, className);
+                cy.importToSelectedClass(selectors.importItem, `${packagesPath}/${testcase.filename}`, selectors.importItemUrl, className, testcase.format);
+                cy.wait(50); // Safe delay for working with FS in Cypress
```
Comment (gabrielfs7) [inline views/cypress/tests/import-export.spec.js:83]: https://github.com/oat-sa/extension-tao-item/pull/572/files#r814812207

If it depends on the environment, we might consider provide this timeout through configuration

File: views/cypress/tests/import-export.spec.js [hunk 3]
```diff
@@ -63,19 +65,31 @@ describe('Import/export items', () => {
...
             );
-        };
-
-        it('can import item', function () {
-            importItemTest('e2e_item.zip');
         });
 
-        it('can import item with shared stimulus', function () {
-            importItemTest('e2e_item_shared_stimulus.zip');
+        [{
```
Comment (gabrielfs7) [inline views/cypress/tests/import-export.spec.js:70]: Where is the CSV import?

Comment (meefox) [reply views/cypress/tests/import-export.spec.js:69]: There is no csv import on the UI

Comment (gabrielfs7) [reply views/cypress/tests/import-export.spec.js:69]: It is in the description of the main ticket. Maybe it was about metadata import only, but we only export it (unless we are talking about tabular import - which is CSV).

**Review by bziondik — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

image

**Review by jsconan — COMMENTED** [review]
The test most likely works, when targeting the remote env.

image

In a local environment, it sometimes fails in timeout. Maybe could we do something to improve this, in a further ticket.

The overall look good. However, I have concerns about the implementation. Please look at my comment below.

**Review by Silvia-loza — APPROVED** [review]
It all looks good and tests pass consistently, well done! :clap:
- [X] New code is covered by tests (if applicable)
- [X] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X]  Pull request's target is not ` master`
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**